internal/dev_server/ui/src/Flags.tsx

```diff
@@ -23,61 +29,70 @@ function Flags({ selectedProject, flags, setFlags }: FlagProps) {
     Record<string, FlagVariation[]>
   >({});
 
-  const overridesPresent = overrides && Object.keys(overrides).length > 0;
+  const overridesPresent = useMemo(
+    () => overrides && Object.keys(overrides).length > 0,
+    [overrides],
+  );
 
-  const updateOverride = (flagKey: string, overrideValue: LDFlagValue) => {
-    const updatedOverrides = {
-      ...overrides,
-      ...{
-        [flagKey]: {
-          value: overrideValue,
+  const updateOverride = useCallback(
```

Comment:
useCallback lets us cache this func definition between renders

Reply:
Do we want that though? We closure in the override state...

Reply:
ohhh, I see you also pass in the deps.